xtokens/src/lib.rs

```diff
@@ -254,6 +345,65 @@ pub mod module {
 			Ok(())
 		}
 
+		fn do_transfer_multiasset_with_fee(
+			who: T::AccountId,
+			asset: MultiAsset,
+			fee: MultiAsset,
+			dest: MultiLocation,
+			dest_weight: Weight,
+			deposit_event: bool,
+		) -> DispatchResult {
+			if !asset.is_fungible(None) || !fee.is_fungible(None) {
+				return Err(Error::<T>::NotFungible.into());
+			}
+
+			if fungible_amount(&asset).is_zero() || fungible_amount(&fee).is_zero() {
```

Comment:
zero fee should be an error?

Reply:
Initially I thought this could be good, but it might prevent some usecases where you transfer to free-execution chains. I will remove it

Reply:
I guess we can revisit this later if there is indeed a chain charges zero fee for buying weights.